Pocket/pocket-ios
# IN-557 + IN-558: My List + Archive Empty States

## Summary

This pull request adds the empty states to our My List and Archive sections. There are 3 empty states considered: 
1. Empty My List
2. Empty Archive List
3. Empty Favorites List 

Along with this addition, there were also a few UI changes, such as fixing the color of the icon when toggling between My List and Archive. Also, fixing the proportions of the star icon next to the Favorite topic chip. 

## References
IN-557
IN-558

## Implementation Details
Overall, this work includes creating a reusable view for the various empty states and this view is added as a collection view cell in our `ItemsListViewController`. We also update the snapshot in `SavedItemsListViewModel` and `ArchivedItemsListViewModel` to add the `emptyState` section if there are no items in the list. 

The `EmptyStateView` was built using SwiftUI and we use the UIHostingController so that this view can act as a collection view cell. The proper view model get passed in (`MyListEmptyStateViewModel`, `ArchiveEmptyStateViewModel`, `FavoritesEmptyStateViewModel`). Future view models that get added will need to add the logic in `emptyState` property of the `ItemsListViewModel`.

Additionally, unit tests were added to `SavedItemsListViewModelTests` and `ArchivedItemsListViewModelTests` and the `EmptyStateTests` for UI tests.

## Testing Steps
1. Navigate to My List
2. Remove all items from the list and verify empty state view appears
3. Tap on button should present "how-to" article 
4. Toggle Favorites button and verify the favorites empty state
5. Perform 1 - 4 for Archive list as well

## Screenshots
![Simulator Screen Shot - iPhone 12 mini - 2022-06-15 at 11 35 14](https://user-images.githubusercontent.com/6743397/173867743-5b437fff-70dc-4afc-9596-7c21639fba89.png)
![Simulator Screen Shot - iPhone 12 mini - 2022-06-15 at 11 34 22](https://user-images.githubusercontent.com/6743397/173867523-e12fdff1-f9c0-4e8c-a7de-830f5850edbb.png)
![Simulator Screen Shot - iPhone 12 mini - 2022-06-15 at 11 34 27](https://user-images.githubusercontent.com/6743397/173867552-6d3c54f5-cdf3-4f9b-a5b0-ede6d6a5e55c.png)
